shipyard_blueprints/snowflake/shipyard_snowflake/snowpark.py

```diff
@@ -0,0 +1,124 @@
+import os
+import snowflake.snowpark.functions as F
+import snowflake.snowpark.types as T
+import pandas as pd
+from snowflake.snowpark.session import Session
+from snowflake.snowpark.window import Window
+from shipyard_templates import ExitCodeException, Database
+
+
+class SnowparkClient(Database):
+    def __init__(
+        self,
+        username,
+        pwd,
+        database=None,
+        account=None,
+        warehouse=None,
+        schema=None,
+        role=None,
+    ):
+        self.username = username
+        self.pwd = pwd
+        self.account = account
+        self.database = database
+        self.warehouse = warehouse
+        self.schema = schema
+        self.role = role
+
+        super().__init__(
+            username,
+            pwd,
+            account=account,
+            warehouse=warehouse,
+            schema=schema,
+            database=database,
+        )
+
+    def connect(self):
```

Comment:
Not sure if it is worth refactoring this to return 1 or 0 to match the other connect method. Also looked at the current authtest- we may want to log the error message there as well so that the end user gets more details

Reply:
For database's and cloud storage, I think it is best if the `connect` method actually returns the connected instance, that is how something like `SQLAlchemy` does it. I think this makes more sense when you need to establish some sort of connection, rather than just call an API